share/p2p/shrexnd/client.go

```diff
@@ -0,0 +1,173 @@
+package shrexnd
+
+import (
+	"errors"
+	"fmt"
+
+	"time"
+
+	"github.com/ipfs/go-cid"
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"golang.org/x/net/context"
+
+	"github.com/celestiaorg/celestia-node/share"
+	"github.com/celestiaorg/celestia-node/share/ipld"
+	share_p2p_v1 "github.com/celestiaorg/celestia-node/share/p2p/shrexnd/v1/pb"
+	"github.com/celestiaorg/go-libp2p-messenger/serde"
+	"github.com/celestiaorg/nmt/namespace"
+)
+
+var (
+	errNotAvailable = errors.New("none of the peers has requested data")
+)
+
+// Client implements Getter interface, requesting data from remote peers
+type Client struct {
+	host           host.Host
+	defaultTimeout time.Duration
```

Comment:
I wouldn't name any struct fields default, only consts

Reply:
renamed to just `timeout`, it is not default tho